### PR TITLE
ENH: Implement Series.drop_duplicates

### PR DESCRIPTION
**Pull request by charlesdong1991**
#874

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -1237,6 +1237,50 @@ def to_list(self):
...
+        Name: animal, dtype: object
+
+        >>> s.drop_duplicates()
+        1       cow
+        0      lama
+        5     hippo
+        3    beetle
+        Name: animal, dtype: object
+        """
+        kdf = self._kdf.drop_duplicates()
```
Comment (ueshin) [inline databricks/koalas/series.py]: `self.to_frame().drop_duplicates()`? 
`self._kdf` doesn't know the current `scol`. Also `self._kdf` might have other columns, then the result could differ.
E.g., consider something like `(s + 1).drop_duplicates()`.

Comment (charlesdong1991) [reply databricks/koalas/series.py]: ehh, thanks! didn't notice this effect. 👍

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -1237,6 +1237,50 @@ def to_list(self):
...
+        Name: animal, dtype: object
+        """
+        kdf = self.to_frame().drop_duplicates()
+        internal = kdf._internal.copy()
+
+        if inplace:
+            self._internal = kdf._internal
+            self._kdf = kdf
+        else:
+            return Series(internal, anchor=kdf)
```
Comment (ueshin) [inline databricks/koalas/series.py]: This loses `internal.scol`, e.g.:

```
>>> s = ks.Series([1, 1, 2, 4, 3])
>>> s.drop_duplicates()
0    1
4    3
2    2
3    4
Name: 0, dtype: int64
>>> (s.drop_duplicates() + 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/base.py", line 143, in __add__
    return _column_op(spark.Column.__add__)(self, other)
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/base.py", line 57, in wrapper
    scol = f(self._scol, *args)
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pyspark/sql/column.py", line 115, in _
    njc = getattr(self._jc, name)(jc)
AttributeError: 'NoneType' object has no attribute '_jc'
```

Comment (charlesdong1991) [reply databricks/koalas/series.py]: thanks a lot, i think i misused this after reading Series definition and implementation. Please take a look if new change is okay.

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -1237,6 +1237,49 @@ def to_list(self):
...
+        0      lama
+        5     hippo
+        3    beetle
+        Name: animal, dtype: object
+        """
+        kseries = _col(self.to_frame().drop_duplicates())
+
+        if inplace:
+            self._internal = kseries._internal
+            self._kdf = kseries
```
Comment (ueshin) [inline databricks/koalas/series.py]: `self._kdf = kseries._kdf`?

Comment (charlesdong1991) [reply databricks/koalas/series.py]: woops, yeah! added! thanks!

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**